File: javaPoker/Card.java
import java.util.*;
//import java.util.Integer;
public class Card
{

// Card class holds identity of card with values. Will be extended to be included in a deck class to add uniqueness to cards and sort them.
    int Face;
    String Suit;


    public Card()
    {
        Face= 0;
        Suit= "";
    }

    public Card(int f, String s)
    {
        Face= f;
        Suit= s;
    }
    
    public Card(String c)
    {
        if(c.length()==3)
        {
            Face= 10;
            Suit=""+c.charAt(2);
        }
        else
        {
            switch(c.charAt(0))
            {
                case 'A':   Face= 1;
                            Suit=""+c.charAt(1);
                            break;
                case 'K':   Face=13;
                            Suit=""+c.charAt(1);
                            break;
                case 'Q':   Face=12;
                            Suit=""+c.charAt(1);
                            break;
                case 'J':   Face=11;
                            Suit=""+c.charAt(1);
                            break;
                default :   Face=(int)c.charAt(0);
                            Suit=""+c.charAt(1);
                            break;
            }

        }


    }

    public int getFace()
    {
        return Face;
    }

    public String getSuit()
    {
        return Suit;
    }

    public void printCard()
    {
        
        switch(Face)
        {
            case 1: System.out.print("A"+Suit+" ");
                    break;
            case 11: System.out.print("J"+Suit+" ");
                    break;
            case 12: System.out.print("Q"+Suit+" ");
                    break;
            case 13: System.out.print("K"+Suit+" ");
                    break;
            default: System.out.print(""+Face+""+Suit+" ");
                    break;
        }
        
    }


}

File: goPoker/card.go
package main



import
(
	"fmt"
	"strconv"
)

 type Card struct 
 {

	Face int
	Suit int

 }

 func newCard(f, s int) Card {
	return Card{Face: f, Suit: s}
}
 
func NewCardFromString(card string) Card {
	face := 0
	switch card[0] {
	case 'A':
		face = 1
	case 'K':
		face = 13
	case 'Q':
		face = 12
	case 'J':
		face = 11
	case '1':
		face = 10
	default:
		face, _ = strconv.Atoi(string(card[0]))
	}

	suit := 0
	switch card[len(card)-1] {
	case 'S':
		suit = 1
	case 'H':
		suit = 2
	case 'C':
		suit = 3
	case 'D':
		suit = 4
	}

	return Card{Face: face, Suit: suit}
}

func (c Card) printCard() {
	switch c.Face {
	case 1:
		fmt.Print("A")
	case 13:
		fmt.Print("K")
	case 12:
		fmt.Print("Q")
	case 11:
		fmt.Print("J")
	case 14:
		fmt.Print("A")
	default:
		fmt.Print(c.Face)
	}

	switch c.Suit {
	case 1:
		fmt.Print("S")
	case 2:
		fmt.Print("H")
	case 3:
		fmt.Print("C")
	case 4:
		fmt.Print("D")
	}
}


